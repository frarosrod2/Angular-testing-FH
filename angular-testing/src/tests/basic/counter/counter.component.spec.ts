import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compile: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CounterComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hacer match', () => {
    expect(compile).toMatchSnapshot();
  });

  test('increaseBy increment', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15);
  });
});
