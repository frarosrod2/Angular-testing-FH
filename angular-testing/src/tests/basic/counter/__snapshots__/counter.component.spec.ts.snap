// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent debe de hacer match 1`] = `undefined`;
